### PR TITLE
Minor uniform performance

### PR DESCRIPTION
**Pull request by pjcozzi**
Improve setting uniform arrays by making fewer WebGL calls, but more expensive ones.  This is also a step closer towards uniform buffers and improved the unit tests for uniforms (some of the oldest code in Cesium btw).

Also improved setting `mat2`, `mat3`, and `mat4` uniforms by making fewer passes over the elements before making the GL call.  This could be tweaked more for the case when the uniform is not set, but I would just wait until we have separate static and dynamic uniforms.

Profile of Mount Everest steady state at 3K:
```
master (% total):

UniformArray.setFloatVec4:   3.85%,  3.61%,  3.53%
Uniform.setMat4:             2.47%,  1.91%,  2.19%

this branch (% total):

UniformArray.setFloatVec4:  3.85%, 3.11%, 2.74%
Uniform.setMat4:            2.51%, 2.47%, 1.99%
```

### CODE REVIEW

**Comment by pjcozzi** [comment]
>This could be tweaked more for the case when the uniform is not set, but I would just wait until we have separate static and dynamic uniforms.

Nevermind, I tweaked this.  It is simpler.  Performance is six of one, half dozen of the other when we are looking at nickles and dimes.

Before (in this branch):
```
Uniform.setMat4:            2.51%, 2.47%, 1.99%
```

After:
```
Uniform.setMat4:            2.08%, 2.45%, 2.65%
```

**Comment by pjcozzi** [comment]
This could also use some testing on mobile devices.

**Comment by mramato** [comment]
@kring, is merging this going to make your life hell for upgrading Jasmine?

**Comment by kring** [comment]
> is merging this going to make your life hell for upgrading Jasmine?

Ha, probably.  My own fault for sitting on it for a month and a half though right?

**Comment by pjcozzi** [comment]
> My own fault for sitting on it for a month and a half though right?

Agreed :)

Changes like that need to be made swiftly.

This is ready now. :)

File: CHANGES.md [hunk 1]
```diff
@@ -4,11 +4,15 @@ Change Log
...
 * Breaking changes
+   *
+* Deprecated
+   *
   * Removed `viewerEntityMixin`, which was deprecated in Cesium 1.5. Its functionality is now directly part of the `Viewer` widget.
   * Removed `Camera.tilt`, which was deprecated in Cesium 1.6. Use `Camera.pitch`.
   * Removed `Camera.heading` and `Camera.tilt`. They were deprecated in Cesium 1.6. Use `Camera.setView`.
   * Removed `Camera.setPositionCartographic`, which was was deprecated in Cesium 1.6. Use `Camera.setView`.
 * Fixed incorrect ellipse texture coordinates. [#2363](https://github.com/AnalyticalGraphicsInc/cesium/issues/2363) and [#2465](https://github.com/AnalyticalGraphicsInc/cesium/issues/2465)
+* Added `pack` and `unpack` functions to `Matrix2`, `Matrix3`, and TODO.
```
Comment (mramato) [inline CHANGES.md]: Looks like this still needs to be updated.

**Comment by mramato** [comment]
Merge in master and fix CHANGES and this should be good to go.

**Comment by pjcozzi** [comment]
Ready.

**Comment by pjcozzi** [comment]
Fixed tests after merge.  This is ready.

**Pull request merged**